File: Python3/프로그래머스/3/12987. 숫자 게임/숫자 게임.py
from collections import deque

def solution(A, B):
    answer = 0
    A.sort()
    B = deque(sorted(B))
    
    score = -1
    for i in A:
        score = -1
        while B and score <= i:
            score = B.popleft()
        if score > i:
            answer += 1
            
    
    return answer

File: C++/프로그래머스/3/12987. 숫자 게임/숫자 게임.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    sort(A.begin(), A.end());
    sort(B.begin(), B.end());
    int answer = 0, i = 0;
    
    for(auto& an : A) {
        while(i < B.size() && B[i] <= an) {
            i++;
        }
        if (i++ == B.size()) {
            break;
        }
        answer++;
    }
    
    
    return answer;
}